scilpy/tractograms/streamline_operations.py

```diff
@@ -388,3 +389,34 @@ def smooth_line_spline(streamline, smoothing_parameter, nb_ctrl_points):
     smoothed_streamline[-1] = streamline[-1]
 
     return smoothed_streamline
+
+
+def generate_matched_points(sft):
+    """
+    Generates an array where each element i is set to the index of the
+    streamline to which it belongs
+
+    Parameters:
+    -----------
+    sft : StatefulTractogram
+        The stateful tractogram containing the streamlines.
+
+    Returns:
+    --------
+    matched_points : ndarray
+        An array where each element is set to the index of the streamline
+        to which it belongs
+
+
+    """
+    total_points = sft.streamlines._data.shape[0]
```

Comment:
With the idea of not using the hidden _data, could you use:
total_points = np.sum([len(s) for s in sft.streamlines])